client/telemetry/src/lib.rs

```diff
@@ -397,6 +504,41 @@ impl TelemetryHandle {
 		}
 	}
 
+	/// Sends the hardware benchmark results through the telemetry connection.
+	///
+	/// This will also register the resulting message as a connection message
+	/// which will be automatically resent in case of a disconnection.
+	///
+	/// Can be called multiple times; any subsequent call will resend the data
+	/// and update the previously registered connection message.
+	///
+	/// Must be called *after* the telemetry has been initialized.
+	pub fn send_hwbench(&self, hwbench: HwBench) {
```

Comment:
Why aren't you using the `on_connect_stream`?

I don't really get why you need to modify the telemetry crate?

Reply:
> I don't really get why you need to modify the telemetry crate?

I don't *need* to per-se; I did it because:

* The new telemetry is effectively part of the protocol since it has special handling on the receiving end of the telemetry, so to me it makes sense to be in the telemetry crate and to make the telemetry crate aware of it.
* The code already has partial support for automatically sending multiple connection messages - the `Node::connection_messages` is a `Vec` so whoever wrote it originally intended for there to be multiple of those, although before before my changes 1) it wasn't actually possible to add new connection messages, and 2) they were sent in the wrong order (but since there was only one connection message it didn't matter).
* It's one less extra task running in the background.

Anyway, I've reverted those changes.